schema-registry/common/src/test/java/com/hortonworks/registries/schemaregistry/retry/RetryExecutorTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2016-2019 Cloudera, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hortonworks.registries.schemaregistry.retry;
+
+import com.hortonworks.registries.schemaregistry.retry.policy.ExponentialBackoffPolicy;
+import com.hortonworks.registries.schemaregistry.retry.policy.FixedTimeBackoffPolicy;
+import com.hortonworks.registries.schemaregistry.retry.policy.BackoffPolicy;
+import com.hortonworks.registries.util.CustomParameterizedRunner;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+@RunWith(CustomParameterizedRunner.class)
+public class RetryExecutorTest {
+
+    enum RetryPolicyType {
+        FIXED,
+        EXPONENTIAL_BACKOFF;
+    }
+
+    private static RetryPolicyType retryPolicyType;
+
+    @CustomParameterizedRunner.Parameters
+    public static Iterable<RetryPolicyType> profiles() {
+        return Arrays.asList(RetryPolicyType.FIXED, RetryPolicyType.EXPONENTIAL_BACKOFF);
+    }
+
+    @CustomParameterizedRunner.BeforeParam
+    public static void beforeParam(RetryPolicyType retryPolicyTypeFromParameter) throws Exception {
+        retryPolicyType = retryPolicyTypeFromParameter;
+    }
+
+    public RetryExecutorTest(RetryPolicyType retryPolicyType) {
+    }
+
+    @Test
+    public void testForMaxIteration() {
+        AtomicInteger iteration = new AtomicInteger(0);
+
+        createRetryExecutor(100, 3, 100000).execute((() -> {
+            iteration.incrementAndGet();
+            if (iteration.get() < 2) {
+                throw new RuntimeException();
+            } else {
+                return null;
+            }
+        }));
+
+        Assert.assertEquals(2, iteration.get());
```

Comment:
maxAttempts is set to 3. Why the expected value in assert statement is 2? (change the iteration to attempt and initialise it to 1).

Reply:
Done